antlr-kotlin-runtime/src/commonMain/kotlin/org/antlr/v4/kotlinruntime/tree/xpath/XPathLexer.kt

```diff
@@ -5,155 +5,155 @@
+
+  override val modeNames: Array<String> =
+    arrayOf("DEFAULT_MODE")
+
+  override val vocabulary: Vocabulary =
+    VocabularyImpl(
+      arrayOf(null, null, null, "'//'", "'/'", "'*'", "'!'"),
+      arrayOf(null, "TOKEN_REF", "RULE_REF", "ANYWHERE", "ROOT", "WILDCARD", "BANG", "ID", "STRING"),
+    )
+
+  override val atn: ATN
+    get() = throw UnsupportedOperationException("Should never access 'atn'")
+
+  override var charPositionInLine: Int = 0
+
+  override fun nextToken(): Token {
+    _tokenStartCharIndex = _input.index()
+    var t: CommonToken? = null
+
+    while (t == null) {
+      when (_input.LA(1)) {
+        '/'.code -> {
+          consume()
+
+          if (_input.LA(1) == '/'.code) {
+            consume()
+            t = CommonToken(ANYWHERE, "//")
+          } else {
+            t = CommonToken(ROOT, "/")
+          }
+        }
+        '*'.code -> {
+          consume()
+          t = CommonToken(WILDCARD, "*")
+        }
+        '!'.code -> {
+          consume()
+          t = CommonToken(BANG, "!")
+        }
+        '\''.code -> {
+          val s = matchString()
+          t = CommonToken(STRING, s)
+        }
+        CharStream.EOF -> {
+          return CommonToken(EOF, "<EOF>")
+        }
+        else -> {
+          if (isNameStartChar(_input.LA(1))) {
+            val id = matchID()
+            t = if (id[0].isUpperCase()) {
+              CommonToken(TOKEN_REF, id)
+            } else {
+              CommonToken(RULE_REF, id)
+            }
+          } else {
+            throw LexerNoViableAltException(this, _input, _tokenStartCharIndex, null)
+          }
+        }
+      }
+    }
+
+    t.startIndex = _tokenStartCharIndex
+    t.charPositionInLine = _tokenStartCharIndex
+    t.line = _line
+    return t
+  }
+
+  public fun consume() {
+    val curChar = _input.LA(1)
+
+    if (curChar == '\n'.code) {
+      _line++
+      charPositionInLine = 0
+    } else {
+      charPositionInLine++
+    }
+
+    _input.consume()
+  }
+
+  public fun matchID(): String {
+    val start = _input.index()
+
+    // Drop start char
+    consume()
+
+    while (isNameChar(_input.LA(1))) {
+      consume()
+    }
+
+    return _input.getText(Interval.of(start, _input.index() - 1))
+  }
+
+  public fun matchString(): String {
+    val start = _input.index()
+
+    // Drop first quote
+    consume()
+
+    while (_input.LA(1) != '\''.code) {
+      consume()
+    }
+
+    // Drop last quote
+    consume()
+    return _input.getText(Interval.of(start, _input.index() - 1))
+  }
+
+  public fun isNameChar(@Suppress("UNUSED_PARAMETER") c: Int): Boolean {
+    TODO("Need isUnicodeIdentifierPart")
+    // return Character.isUnicodeIdentifierPart(c)
+  }
+
+  public fun isNameStartChar(@Suppress("UNUSED_PARAMETER") c: Int): Boolean {
+    TODO("Need isUnicodeIdentifierStart")
+    // return Character.isUnicodeIdentifierStart(c)
+  }
```

Comment:
If you know how to re-implement those two methods, feel free to let me know or open another PR after this one is merged.